desc/magnetic_fields.py

```diff
@@ -1094,3 +1130,644 @@ def odefun(rpz, s):
+            the form potential(theta,zeta,**params) -> ndarray.
+            theta,zeta are poloidal and toroidal angles on the surface
+        potential_dtheta: callable
+            function to compute the theta derivative of the current potential
+        potential_dzeta: callable
+            function to compute the theta derivative of the current potential
+        surface_grid : Grid,
+            grid upon which to evaluate the surface current density K
+        params : dict, optional
+            default parameters to pass to potential function (and its derivatives)
+
+        name : str
+            name for this field
+        check_orientation : bool
+            ensure that this surface has a right handed orientation. Do not set to False
+            unless you are sure the parameterization you have given is right handed
+            (ie, e_theta x e_zeta points outward from the surface).
+
+        """
+        if not isinstance(surface, FourierRZToroidalSurface):
+            raise TypeError(
+                "Expected type FourierRZToroidalSurface for argument surface, "
+                f"instead got type {type(surface)}"
+            )
+        R_lmn = surface.R_lmn
+        Z_lmn = surface.Z_lmn
+        modes_R = surface._R_basis.modes[:, 1:]
+        modes_Z = surface._Z_basis.modes[:, 1:]
+        NFP = surface.NFP
+        sym = surface.sym
+        rho = surface.rho
+
+        return cls(
+            potential,
+            potential_dtheta,
+            potential_dzeta,
+            surface_grid,
+            params,
+            R_lmn,
+            Z_lmn,
+            modes_R,
+            modes_Z,
+            NFP,
+            sym,
+            rho,
+            name,
+            check_orientation,
+        )
+
+
+class FourierCurrentPotentialField(_MagneticField, FourierRZToroidalSurface):
+    """Magnetic field due to a surface current potential on a toroidal surface.
+
+        surface current K is assumed given by
+
+        K = ∇ Φ x n
+        Φ(θ,ζ) = Φₛᵥ(θ,ζ) + Gζ/2π + Iθ/2π
+
+        where:
+              n is the winding surface unit normal
+              Phi is the current potential function,
+              which is a function of theta and zeta,
+              and is given as a secular linear term in theta/zeta
+              and a double Fourier series in theta/zeta
+        this function then uses biot-savart to find the
+        B field from this current density K on the surface
+
+    Parameters
+    ----------
+    Phi_mn : ndarray
+        Fourier coefficients of the double FourierSeries part of the current potential.
+    modes_Phi : array-like, shape(k,2)
+        Poloidal and Toroidal modenumbers corresponding to passed-in Phi_mn coefficients
+    I : float
+        Net current linking the plasma and the coils toroidally
+        Denoted I in the algorithm
+    G : float
+        Net current linking the plasma and the coils poloidally
+        Denoted G in the algorithm
+    surface_grid : Grid
```

Comment:
is this needed here, or just when we compute things?

Reply:
mainly when we compute things, BUT I want this class to be like the other MagneticField classes in that it defines a single magnetic field, in this case one given by a surface current, and so I want it to have as attributes the `I` and `G`